clients/src/main/java/org/apache/kafka/common/utils/Utils.java

```diff
@@ -371,6 +371,7 @@ public static boolean isEqualConstantTime(char[] first, char[] second) {
             int j = i < second.length ? i : 0;
             if (first[i] != second[j]) {
                 matches = false;
+                break;
```

Comment:
This is expected behavior. see the comment `time-constant comparison that always compares all characters in first array`

Reply:
Thanks for pointing that out!